homeassistant/components/cover/zwave.py

```diff
@@ -0,0 +1,177 @@
+"""
+Support for Zwave roller shutter components.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/rollershutter.zwave/
+"""
+# Because we do not compile openzwave on CI
+# pylint: disable=import-error
+import logging
+from homeassistant.components.cover import DOMAIN
+from homeassistant.components.zwave import ZWaveDeviceEntity
+from homeassistant.components import zwave
+from homeassistant.components.cover import CoverDevice
+
+COMMAND_CLASS_SWITCH_MULTILEVEL = 0x26  # 38
+COMMAND_CLASS_SWITCH_BINARY = 0x25  # 37
+
+SOMFY = 0x47
+SOMFY_ZRTSI = 0x5a52
+SOMFY_ZRTSI_CONTROLLER = (SOMFY, SOMFY_ZRTSI)
+WORKAROUND = 'workaround'
+
+DEVICE_MAPPINGS = {
+    SOMFY_ZRTSI_CONTROLLER: WORKAROUND
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Find and return Z-Wave roller shutters."""
+    if discovery_info is None or zwave.NETWORK is None:
+        return
+
+    node = zwave.NETWORK.nodes[discovery_info[zwave.ATTR_NODE_ID]]
+    value = node.values[discovery_info[zwave.ATTR_VALUE_ID]]
+
+    if (value.command_class == zwave.COMMAND_CLASS_SWITCH_MULTILEVEL and
+            value.index == 0):
+        value.set_change_verified(False)
+        add_devices([ZwaveRollershutter(value)])
+    elif (value.command_class == zwave.COMMAND_CLASS_SWITCH_BINARY or
+          value.command_class == zwave.COMMAND_CLASS_BARRIER_OPERATOR):
+        if value.type != zwave.TYPE_BOOL:
+            return
+        if value.genre != zwave.GENRE_USER:
+            return
+        value.set_change_verified(False)
+        add_devices([ZwaveGarageDoor(value)])
+    else:
+        return
+
+
+class ZwaveRollershutter(zwave.ZWaveDeviceEntity, CoverDevice):
```

Comment:
For platforms who needs to provide both garage door and rollershutter specific code, I think it will make sense to differentiate between their names.